plugins/modules/storagebox_subaccount.py

```diff
@@ -0,0 +1,449 @@
+
+subaccount:
+  description: The subaccount object returned by the API.
+  type: dict
+  returned: if O(state=present)
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+from ansible_collections.community.hrobot.plugins.module_utils.robot import (
+    BASE_URL,
+    ROBOT_DEFAULT_ARGUMENT_SPEC,
+    fetch_url_json,
+)
+
+
+def encode_data(data):
+    """Converts booleans to lowercase strings and filters out None values."""
+    return urlencode(
+        {
+            key: str(value).lower() if isinstance(value, bool) else value
+            for key, value in data.items()
+            if value is not None
+        }
+    )
+
+
+def create_subaccount(module, storagebox_id, subaccount):
+    url = "{0}/storagebox/{1}/subaccount".format(BASE_URL, storagebox_id)
+    res, error = fetch_url_json(
+        module,
+        url,
+        method="POST",
+        data=encode_data(subaccount),
+        headers={"Content-type": "application/x-www-form-urlencoded"},
+        accept_errors=[
+            "STORAGEBOX_SUBACCOUNT_LIMIT_EXCEEDED",
+            "STORAGEBOX_INVALID_PASSWORD",
+        ],
+        timeout=30000,  # this endpoint is stupidly slow
+    )
+
+    if error == "STORAGEBOX_INVALID_PASSWORD":
+        module.fail_json(msg="Invalid password (says Hetzner)")
+    if error == "STORAGEBOX_SUBACCOUNT_LIMIT_EXCEEDED":
+        module.fail_json(msg="Subaccount limit exceeded")
+
+    # Contains all subaccount informations
+    # { "subaccount": <data> }
+    return res["subaccount"]
+
+
+def merge_subaccounts_infos(original, updates):
+    # None values aren't updated
+    result = original.copy()
+    for key, value in updates.items():
+        if value is not None:
+            result[key] = value
+    return result
+
+
+def is_subaccount_updated(before, after):
+    for key, value in after.items():
+        # Means user didn't provide a value
+        # we assume we don't want to update that field
+        if value is None:
+            continue
+        # password aren't considered part of update check
+        # due to being a different API call
+        if key == "password":
+            continue
+        if before.get(key) != value:
+            return True
+    return False
+
+
+def delete_subaccount(module, storagebox_id, subaccount):
+    empty, error = fetch_url_json(
+        module,
+        "{0}/storagebox/{1}/subaccount/{2}".format(
+            BASE_URL, storagebox_id, subaccount["username"]
+        ),
+        method="DELETE",
+        allow_empty_result=True,
+        headers={"Content-type": "application/x-www-form-urlencoded"},
+    )
+
+
+def update_subaccount(module, storagebox_id, subaccount):
+    empty, error = fetch_url_json(
+        module,
+        "{0}/storagebox/{1}/subaccount/{2}".format(
+            BASE_URL, storagebox_id, subaccount["username"]
+        ),
+        method="PUT",
+        data=encode_data(subaccount),
```

Comment:
Here I wouldn't provide the `password` field, to avoid accidental leaks of the password:
```suggestion
        data=encode_data({key: value for key, value in subaccount.items() if key != "password"}),
```